presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java

```diff
@@ -319,6 +320,77 @@ public void testSpatialJoin()
                                 values("RIGHT_SYMBOL"))));
     }
 
+    @Test
+    public void testSemiJoin()
+    {
+        // Unconsumed dynamic filter
+        PlanNode root = builder.semiJoin(
+                builder.filter(expression("ORDERS_OK > 0"), ordersTableScanNode),
+                lineitemTableScanNode,
+                ordersOrderKeySymbol,
+                lineitemOrderKeySymbol,
+                new Symbol("SEMIJOIN_OUTPUT"),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new DynamicFilterId("DF")));
+        assertPlan(
+                removeUnsupportedDynamicFilters(root),
+                semiJoin("ORDERS_OK", "LINEITEM_OK", "SEMIJOIN_OUTPUT", Optional.empty(), false,
+                        filter("ORDERS_OK > 0",
```

Comment:
use `io.prestosql.sql.planner.assertions.PlanMatchPattern#filter(io.prestosql.sql.tree.Expression, io.prestosql.sql.tree.Expression, io.prestosql.sql.planner.assertions.PlanMatchPattern)` version of filter to assert DF is really gone